system/Database/OCI8/PreparedQuery.php

```diff
@@ -74,11 +74,21 @@ public function _execute(array $data): bool
         }
 
         foreach (array_keys($data) as $key) {
-            oci_bind_by_name($this->statement, ':' . $key, $data[$key]);
+            if (is_string($data[$key]) && $this->isBinary($data[$key])) {
+                $binaryData = oci_new_descriptor($this->db->connID, OCI_D_LOB);
+                $binaryData->writeTemporary($data[$key], OCI_TEMP_BLOB);
+                oci_bind_by_name($this->statement, ':' . $key, $binaryData, -1, OCI_B_BLOB);
+            } else {
+                oci_bind_by_name($this->statement, ':' . $key, $data[$key]);
+            }
         }
 
         $result = oci_execute($this->statement, $this->db->commitMode);
 
+        if (isset($binaryData)) {
```

Comment:
how about add 

```php
$binaryData = null;
```

initialization before foreach, so the check can be updated to:

```php
if ($binaryData instanceof \OCILob)
```